pure2.md

```diff
@@ -0,0 +1,535 @@
+
+3.  Update offline-update Snapshot metadata from the well-known location:
+
+    1.  Load the offline-update Snapshot metadata file from the well-known location. If the version number of the existing offline-update Snapshot metadata file is equal to or greater than the one found in the well-known location, continue to step 4.
+
+    2.  Otherwise, check that it has been signed by the threshold of unique keys specified for the offline-update Snapshot role in the latest Root metadata file. If the new Snapshot metadata file is not signed as required, discard it, abort the update cycle, and log the signature failure. (Checks for an arbitrary software attack.)
+
+    3.  Check that the version number listed by the previous Snapshot metadata file for each offline-update Targets metadata file is less than or equal to its version number in this Snapshot metadata file. If this condition is not met, discard the new Snapshot metadata file, abort the update cycle, and log the failure. (Checks for a rollback attack.)
+
+    4.  Check that the current (or latest securely attested) time is lower than the expiration timestamp in this Snapshot metadata file. If the new Snapshot metadata file is expired, discard it, abort the update cycle, and log the potential freeze attack. (Checks for a freeze attack.)
+
+4.  Traverse the list of offline-update targets metadata files in the offline-update Snapshot metadata until a matching filename is found in the well-known location. If no matching file is found, abort the update cycle and log the error. If one is found, perform the following verification procedure:
+
+    1.  Check that the version number of this Targets metadata file matches the version number listed in the latest Snapshot metadata. If the version number does not match, discard it, abort the update cycle, and log the failure.
+
+    2.  Check that the Targets metadata has been signed by the threshold of unique keys specified for the offline-updates Targets role in the latest Root metadata file. If not, discard it, abort the update cycle, and log the failure. (Checks for an arbitrary software attack.)
+
+    3.  Check that the current (or latest securely attested) time is lower than the expiration timestamp in this Targets metadata file. If not, abort the update cycle and log the failure. (Checks for a freeze attack.)
+
+    4.  Check that there is at most one target in the metadata that can be installed on this ECU. The exact method for this will vary between implementations. For example, if each target specified a hardwareId, make sure that there is only one target listed for this ECU's hardwareId. If there is more than one target that could be installed on this ECU, abort the update cycle and log the failure (if this is the primary ECU) or report the failure to the primary ECU (if this is a secondary ECU).
```

Comment:
```suggestion
    4.  Check that there is at most one target in the metadata that can be installed on this ECU. The exact method for this will vary between implementations. For example, if each target specified a hardware identifier, make sure that there is only one target listed for this ECU's hardware identifier. If there is more than one target that could be installed on this ECU, abort the update cycle and log the failure (if this is the primary ECU) or report the failure to the primary ECU (if this is a secondary ECU).
```